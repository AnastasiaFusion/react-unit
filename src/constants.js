const letterSize = 12; // average approximate letter width
export const padding = 15;

export const items = [{
    title: 'Home',
    minWidth: letterSize * 4,
  },
  {
    title: 'About',
    minWidth: letterSize * 5,
  },
  {
    title: 'Best Deals',
    minWidth: letterSize * 9,
  },
  {
    title: 'Coupons',
    minWidth: letterSize * 7,
  },
  {
    title: 'Discounts',
    minWidth: letterSize * 9,
  },
  {
    title: 'Contact',
    minWidth: letterSize * 7,
  },
  {
    title: 'Log in',
    minWidth: letterSize * 6,
  },
  {
    title: 'News',
    minWidth: letterSize * 4,
  },
];

export const fullMenuSize = (() => {
  const sizes = items.map(item => item.minWidth + padding * 2);
  return sizes.reduce((acc, curr) => acc + curr)
})();
