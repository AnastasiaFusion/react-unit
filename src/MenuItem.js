import React from 'react';
import styled from 'styled-components';
import { padding } from './constants';

const MenuItem = (props) => {
  const {minWidth, children } = props;

  return (
    <Item minWidth={minWidth} padding={padding}>
      {children}
    </Item>
  );
};

const Item = styled.div`
  min-width: ${({ minWidth }) => `${minWidth}px`};
  color: #2C86EB;
  font-weight: bold;
  height: 40px;
  padding: ${({ padding }) => `5px ${padding}px`};
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;

  &:hover {
    background-color: #D1E6FA;
  }
`;

export default MenuItem;
