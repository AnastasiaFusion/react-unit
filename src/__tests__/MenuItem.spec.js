import React from 'react';
import { shallow, mount } from 'enzyme';
import MenuItem from '../MenuItem';
import { items } from '../constants';

const setUp = props => shallow(<MenuItem {...props} />)
const setMount = props => mount(<MenuItem {...props} />)

describe("MenuItem component", () => {
  let component;

  afterEach(() => {
    component.unmount();
  });

  it("should render MenuItem component", () => {
    const { minWidth, title: children } = items[0];
    component = setUp({ minWidth, children });
    expect(component).toMatchSnapshot();
  })

  it("should include 'About' as text content", () => {
    const { minWidth, title: children } = items[1];
    component = setMount({ minWidth, children });
    expect(component.getDOMNode()).toHaveTextContent('About')
  })
})