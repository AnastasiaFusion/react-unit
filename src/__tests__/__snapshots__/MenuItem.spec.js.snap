// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render MenuItem component 1`] = `
<styled.div
  minWidth={48}
  padding={15}
>
  Home
</styled.div>
`;
